pkg/controllers/machinesync/machine_sync_controller_test.go

```diff
@@ -192,7 +194,11 @@ var _ = Describe("With a running MachineSync Reconciler", func() {
 		mgr, err = ctrl.NewManager(controllerCfg, ctrl.Options{
 			Scheme: testScheme,
 			Controller: config.Controller{
-				SkipNameValidation: ptr.To(true),
+				MaxConcurrentReconciles: 3,
+				SkipNameValidation:      ptr.To(true),
+			},
+			Cache: cache.Options{
+				SyncPeriod: ptr.To(1 * time.Second),
```

Comment:
Requiring `SyncPeriod` is (in general) an anti-pattern. It means we're missing state changes when they happen. This is going to leave customers waiting so long for the thing to reconcile that they start doing rash things like deleting VMs.

What breaks when we remove this? Can we fix it in code?